doc/user/bgp.rst

```diff
@@ -1592,32 +1684,23 @@ Large Communities in Route Map
    overwriting other values. Multiple large-community values can be specified.
 
 
-.. _bgp-vrfs:
-
-VRFs
-----
+.. _bgp-l3vpn-vrfs:
 
-BGP supports multiple VRF instances with the following command:
+L3VPN VRFs
+----------
 
-.. index:: router bgp ASN vrf VRFNAME
-.. clicmd:: router bgp ASN vrf VRFNAME
+*bgpd* supports :abbr:`L3VPN (Layer 3 Virtual Private Networks)` :abbr:`VRFs
+(Virtual Routing and Forwarding)` for IPv4 :rfc:`4364` and IPv6 :rfc:`4659`.
+L3VPN routes, and their associated VRF MPLS labels, can be distributed to VPN
+SAFI neighbors in the *default*, i.e., non VRF, BGP instance. VRF MPLS labels
+are reached using *core* MPLS labels which are distributed using LDP or BGP
+labeled unicast.  *bgpd* also supports inter-VRF route leaking.
 
-``VRFNAME`` is matched against VRFs configured in the kernel. When
-``vrf VRFNAME`` is not specified, the BGP protocol process belongs to the
-default VRF.
 
-With VRF, you can isolate networking information. Having BGP VRF allows you to
-have several BGP instances on the same system process. This solution solves
-scalabiliy issues where the network administrator had previously to run
-separately several BGP processes on each namespace. Now, not only BGP VRF
-solves this, but also this method applies to both kind of VRFs backend: default
-VRF from Linux kernel or network namespaces. Also, having separate BGP
-instances does not imply that the AS number has to be different. For internal
-purposes, it is possible to do iBGP peering from two differents network
-namespaces.
```

Comment:
above paragraph should be moved ( not deleted) into "Multiple Autonomous Systems".
Whatever, pull request #2344 should be kept.

Reply:
>With VRF, you can isolate networking information.

This is explained in Zebra's VRF docs.

>Having BGP VRF allows you to have several BGP instances on the same system process.

Already explained in the first part of the section on Multiple ASs

> This solution solves scalabiliy issues where the network administrator had previously to run separately several BGP processes on each namespace.

Sure, this information could be useful, I don't have a problem keeping this.

>Now, not only BGP VRF solves this, but also this method applies to both kind of VRFs backend: default VRF from Linux kernel or network namespaces.

This is more like trivia, since namespace -vs- real vrf's is a distinction that should be made in the dedicated section on vrf's and not elsewhere (imo). In other words I think it goes without saying.

>Also, having separate BGP instances does not imply that the AS number has to be different.

This is good to note, I'll add it back to the section on Multiple ASs

>For internal purposes, it is possible to do iBGP peering from two differents network namespaces.

I'm not sure this is relevant given the existing information on how VRFs work in BGP, it's more or less a given

Reply:
I agree with the above, except for the iBGP peering from two different network namespaces. 
I think you can add like below:
````
Also, having separate BGP instances does not imply that the AS number has to be different.
Local BGP instances can also be used to do iBGP peering between each instance.
```